examples/threading/threading.c

```diff
@@ -10,16 +10,34 @@
 
 void* threadfunc(void* thread_param)
 {
-
+	
+	
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    //Wait 
+    usleep(1000*thread_func_args->wait_to_obtain_ms);
+    
+    //locking the mutex
+    pthread_mutex_lock(thread_func_args->mutex);
+    
+    //wait
+    usleep(1000*thread_func_args->wait_to_release_ms);
```

Comment:
@swapnilghonge can yo be sure usleep actually waited the full period of time here without checking the return code?